pyQuARC/code/schema_validator.py

```diff
@@ -35,19 +47,60 @@ def __init__(
         # Temporarily set the environment variable
         os.environ['XML_CATALOG_FILES'] = os.environ.get('XML_CATALOG_FILES', catalog_path)
 
-        self.metadata_format = metadata_format
-        self.xml_schema = self.read_xml_schema()
-
-    def read_xml_schema(self):
-        """
-        Reads the xml schema file
-        """
-        with open(SCHEMA_PATHS[f"{self.metadata_format}_xml"], "r") as schema_file:
+        with open(SCHEMA_PATHS[f"{self.metadata_format}_schema"], "r") as schema_file:
             file_content = schema_file.read().encode()
         xmlschema_doc = etree.parse(BytesIO(file_content))
         schema = etree.XMLSchema(xmlschema_doc)
         return schema
 
+    def read_json_schema(self):
+        """
+        Reads the json schema file
+        """
+        schema = json.load(open(SCHEMA_PATHS[f"{self.metadata_format}_schema"], "r"))
+        return schema
+
+    def run_json_validator(self, content_to_validate):
+        """
+        Validate passed content based on the schema and return any errors
+        Args:
+            content_to_validate (str): The metadata content as a json string
+        Returns:
+            (dict) A dictionary that gives the validity of the schema and errors if they exist
+        """
+        schema = self.read_json_schema()
+        schema_base = json.load(open(SCHEMA_PATHS["umm-cmn-json-schema"], "r"))
+
+        # workaround to read local referenced schema file (only supports uri)
+        schema_store = {
+            schema_base.get('$id','/umm-cmn-json-schema.json') : schema_base,
+            schema_base.get('$id','umm-cmn-json-schema.json') : schema_base,
+        }
+
+        errors = {}
+
+        resolver = RefResolver.from_schema(schema, store=schema_store)
+
+        validator = Draft7Validator(
```

Comment:
What is Draft7Validator?

Reply:
[Draft 7](https://json-schema.org/draft-07/json-schema-release-notes.html) is the json schema version (newest) and Draft7Validator is the validator that version.